clang/include/clang/AST/DeclCXX.h

```diff
@@ -1395,7 +1395,8 @@ class CXXRecordDecl : public RecordDecl {
   bool allowConstDefaultInit() const {
     return !data().HasUninitializedFields ||
            !(data().HasDefaultedDefaultConstructor ||
-             needsImplicitDefaultConstructor());
+             needsImplicitDefaultConstructor()) ||
+           (isUnion() && isEmpty());
```

Comment:
Er, why do we special-case empty unions here? Shouldn’t that just work automatically because an empty union has no fields at all and therefore no uninitialised fields either?